tools/python/boutiques/localExec.py

```diff
@@ -1155,6 +1178,118 @@ def check(keyname, isGood, msg, value):  # Checks input values
                 message += ("\t" + err + "\n")
             raise_error(ExecutorError, message)
 
+    # Private method to generate summary object of data
+    # collection file containing the tool name and descriptor DOI
+    def _generateSummary(self, desc):
+        summary = {}
+        summary['name'] = self.desc_dict['name']
+        summary['descriptor-doi'] = self._findDOI(desc)
+        return summary
+
+    # Private method to attempt to find descriptor DOI
+    # through various cases
+    # userInput may be json string, filename or zenodo id
+    def _findDOI(self, userIn):
+        # DOI in Zenodo reference case
+        if userIn.split(".")[0].lower() == "zenodo":
+            return userIn
+        # File cases
+        if os.path.isfile(userIn):
+            # Most recent DOI in file if user is publisher
+            # Include check to ensure descriptor is unmodified
+            if self.desc_dict.get('doi') is not None:
+                doi = self.desc_dict.get('doi')
```

Comment:
You could create variable doi anyway, and then just test if it's None, it would save one test and improve readibility.